pkg/dependency/parser/python/uv/parse.go

```diff
@@ -0,0 +1,160 @@
+package uv
+
+import (
+	"sort"
+
+	"github.com/BurntSushi/toml"
+	"github.com/samber/lo"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/dependency"
+	ftypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+	xio "github.com/aquasecurity/trivy/pkg/x/io"
+)
+
+type Lock struct {
+	Packages []Package `toml:"package"`
+}
+
+func (l Lock) packages() map[string]Package {
+	return lo.SliceToMap(l.Packages, func(pkg Package) (string, Package) {
+		return pkg.Name, pkg
+	})
+}
+
+func (l Lock) directDeps(root Package) map[string]struct{} {
+	deps := make(map[string]struct{})
+	for _, dep := range root.Dependencies {
+		deps[dep.Name] = struct{}{}
+	}
+	return deps
+}
+
+func (l Lock) prodDeps(root Package) map[string]struct{} {
+	packages := l.packages()
```

Comment:
Same as for root package.
We already got `packages` map:
https://github.com/aquasecurity/trivy/blob/e67015d91822dcf2c5839e799f348515827a2b7a/pkg/dependency/parser/python/uv/parse.go#L104
Let's use this map.